File: 633. Sum of Square Numbers/judgeSquareSum.py
from math import sqrt

class Solution:
  def judgeSquareSum(self, c: int) -> bool:
    left = 0
    right = int(sqrt(c))

    while left <= right:
      sum = left * left + right * right
      if sum == c:
        return True
      elif sum < c:
        left += 1
      else:
        right -= 1

    return False


c = 5
solution = Solution()
print('judgeSquareSum: ', solution.judgeSquareSum(c))

File: 633. Sum of Square Numbers/judgeSquareSum.js
/**
 * @param {number} c
 * @return {boolean}
 */
const judgeSquareSum = (c) => {
  let left = 0;
  let right = Math.round(Math.sqrt(c));

  while (left < right) {
    const sum = left * left + right * right;
    if (sum === c) {
      return true;
    } else if (sum < c) {
      left++;
    } else {
      right--;
    }
  }

  return false;
};


const c = 5;
console.log('judgeSquareSum: ', judgeSquareSum(c));
